lib/global_uid/base.rb

```diff
@@ -172,14 +181,45 @@ def self.get_many_uids_for_class(klass, count)
       return [] unless count > 0
       with_connections do |connection|
         Timeout.timeout(self.global_uid_options[:query_timeout], TimeoutException) do
-          increment_by = connection.select_value("SELECT @@auto_increment_increment")
+          increment_by = fetch_and_validate_increment!(connection)
           start_id = connection.insert("REPLACE INTO #{klass.global_uid_table} (stub) VALUES " + (["('a')"] * count).join(','))
           return start_id.step(start_id + (count-1) * increment_by, increment_by).to_a
         end
       end
       raise NoServersAvailableException, "All global UID servers are gone!"
     end
 
+    def self.validate_increment!(new_record_id, connection)
+      server = self.servers.find { |s| connection.current_database.include?(s[:name]) }
+
+      begin
+        unless server[:allocated_ids] << new_record_id
+          raise InvalidIncrementException, \
+            "Recently allocated IDs: #{server[:allocated_ids]} on '#{connection.current_database}'"
+        end
+        return new_record_id
+      rescue InvalidIncrementException => e
```

Comment:
Doesn't seem like this `rescue` ever gets hit with the current code
(Sorry if this still wasn't ready for review)